tests/usm/usm_allocations_helper.h

```diff
@@ -0,0 +1,204 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides different tools for usm allocation/free/usm_allocator tests
+//
+*******************************************************************************/
+#include "../common/common.h"
+#include "usm.h"
+#include <cstddef>
+
+namespace usm_alloc_help {
+using namespace sycl_cts;
+
+/** @brief Custom type to create usm_allocator with other T type
+ */
+struct custom_type {
+  int value;
+};
+
+/** @brief Kernel index to differentiate kernel names in methods
```

Comment:
```suggestion
/** @brief Kernel index to differentiate kernel names in member functions
```
sounds more C++-friendly and actually the `enum` name is about `memb_func` :-)

Reply:
> sounds more C++-friendly and actually the enum name is about memb_func :-)

Agree, thanks, fixed

> It is not clear to me whether you test that the shared & host USM can be used on the host in normal code, outside of a kernel running on a host device.

Actually **ptr** is used on host side (line 160)
https://github.com/KhronosGroup/SYCL-CTS/blob/cee881cf0b30dda339b9c2cc376fc3b2369eb7b8/tests/usm/usm_allocations_helper.h#L154-L160

and line 178
https://github.com/KhronosGroup/SYCL-CTS/blob/cee881cf0b30dda339b9c2cc376fc3b2369eb7b8/tests/usm/usm_allocations_helper.h#L176-L180

Reply:
Ah, yes. Thanks.